sphinx/addnodes.py

```diff
@@ -174,6 +174,31 @@ class desc_content(nodes.General, nodes.Element):
     """
 
 
+class desc_sig_element(nodes.inline):
```

Comment:
I added desc_sig_element and children to realize pygments based inline text decorator (see #4289).
But I also added `SigElementFallbackTransform` to support old translators. It helps to render these nodes as simple inline node in old translators.
It means this does not break compatibility! I confirmed working fine with docxbuilder on my local.

Reply:
Excellent!